conformance/README.md

```diff
@@ -607,33 +607,51 @@ expectation, ending that branch of the test tree.
 These rules describe Ion data-model results for use in the `denotes` expectation:
 
 ```ebnf
-model-value   ::=  model-content  |  annotated
-
-model-content ::=  null.null
-                |  bool
-                |  int
-                |  string
-                |  "("  "null"    model-type    ")"
-                |  "("  "string"  codepoint*    ")"
-                |  "("  "symbol"  model-symtok  ")"
-                |  "("  "list"    model-value*  ")"
-                |  "("  "sexp"    model-value*  ")"
-                |  "("  "struct"  model-field*  ")"
-                |  "("  "blob"    bytes*        ")"
-                |  "("  "clob"    bytes*        ")"
+model-value     ::=  model-content  |  annotated
+
+model-content   ::=  null.null
+                  |  bool
+                  |  int
+                  |  string
+                  |  "("  "null"      model-type       ")"
+                  |  "("  "bool"      bool             ")"
+                  |  "("  "int"       int              ")"
+                  |  "("  "float"     string           ")" // See https://amazon-ion.github.io/ion-docs/docs/float.html
+                  |  "("  "decimal"   model-decimal    ")"
+                  |  "("  "timestamp" model-timestamp  ")"
+                  |  "("  "string"    codepoint*       ")"
+                  |  "("  "symbol"    model-symtok     ")"
+                  |  "("  "list"      model-value*     ")"
+                  |  "("  "sexp"      model-value*     ")"
+                  |  "("  "struct"    model-field*     ")"
+                  |  "("  "blob"      bytes*           ")"
+                  |  "("  "clob"      bytes*           ")"
 
 // TODO Other types per denotational semantics
 
-codepoint     ::=  int                               // in the range 0..0x10FFFF
+codepoint       ::=  int                                   // in the range 0..0x10FFFF
 
-model-symtok  ::=  string
-                |  int
-                |  "("  "text"  codepoint*  ")"
-                |  "("  "absent"  string  int  ")"       // symtab name + offset
+model-symtok    ::=  string
+                  |  int
+                  |  "("  "text"  codepoint*  ")"
+                  |  "("  "absent"  string  int  ")"       // symtab name + offset
 
-model-field   ::=  "("  model-symtok  model-value  ")"
+model-field     ::=  "("  model-symtok  model-value  ")"
 
-annotated     ::=  "("  "annot"  model-content  string*  ")"
+model-decimal   ::= int int                                // coefficient + exponent
+                  | "negative_zero" int
+
+model-timestamp ::= int
+                  | int int
```

Comment:
I'm trying to think through whether the alternate forms here are net-positive.  My instinct is that the model here should reflect all possible fields explicitly; otherwise we are encoding those expectations within the test framework implementation, which could hide/shift defects.

There's also a subtle thing we should cover: the distinction between _precision_ and _point-in-time_.  For example, "2007T" has year-precision, but has the same point-in-time as "2007-01-01T00:00:00Z" and its local offset is unknown.  So we should be able to assert that those two are same point-in-time but _not_ equivalent in the data model.  Probably means we need a special expectation?

I haven't implemented timestamp in the semantic model, so thinking on the fly here... First idea how to handle it is to model each field explicitly to reflect point-in-time, and have a separate precision marker of some kind.  Local offset would need to reflect that both "Z" and "-00:00" are the same offset, but distinct in the data model.

But ouch: local offset handling might make point-in-time comparisons very painful since the other fields would need to be adjusted using calendar math...

There's enough complexity here that I feel like we should pursue this independently of the other types, so we don't block progress elsewhere.